sdk/identity/azure-identity/src/workload_identity_credential.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "azure/identity/workload_identity_credential.hpp"
+
+#include "private/tenant_id_resolver.hpp"
+#include "private/token_credential_impl.hpp"
+
+#include <fstream>
+#include <streambuf>
+
+using Azure::Identity::WorkloadIdentityCredential;
+
+using Azure::Core::Context;
+using Azure::Core::Url;
+using Azure::Core::Credentials::AccessToken;
+using Azure::Core::Credentials::TokenRequestContext;
+using Azure::Core::Http::HttpMethod;
+using Azure::Identity::_detail::TenantIdResolver;
+using Azure::Identity::_detail::TokenCredentialImpl;
+
+WorkloadIdentityCredential::WorkloadIdentityCredential(
+    std::string tenantId,
+    std::string const& clientId,
+    std::string const& tokenFilePath,
+    std::string const& authorityHost,
+    std::vector<std::string> additionallyAllowedTenants,
+    Core::Credentials::TokenCredentialOptions const& options)
+    : TokenCredential("WorkloadIdentityCredential"),
+      m_clientCredentialCore(tenantId, authorityHost, additionallyAllowedTenants),
+      m_tokenFilePath(tokenFilePath),
```

Comment:
[Prefer initializer syntax over method syntax.](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-list).

Reply:
I am following what we are doing elsewhere in identity, such as `client_secret_credential`.

For syntax improvements like these to follow guidelines consistently, I would prefer to fix them across the board (at least for a given package), rather than have piecemeal inconsistency. Is that fair?

Reply:
My personal preference is to always use best practices rather than rely on someone coming along afterwards and cleaning things up. 

But I'm not going to fight on this one.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/4873